content/custom_resources/glossary.md

```diff
@@ -0,0 +1,578 @@
+
+root_context
+
+## property_is_set?
+
+Use the `property_is_set?` method to check if the value for a property is set. The syntax is:
+
+```ruby
+property_is_set?(:property_name)
+```
+
+The `property_is_set?` method will return `true` if the property is set.
+
+For example, the following custom resource creates and/or updates user
+properties, but not their password. The `property_is_set?` method checks
+if the user has specified a password and then tells Chef Infra Client
+what to do if the password is not identical:
+
+```ruby
+action :create do
+  converge_if_changed do
+    shell_out!("rabbitmqctl create_or_update_user #{username} --prop1 #{prop1} ... ")
+  end
+
+  if property_is_set?(:password)
+    if shell_out("rabbitmqctl authenticate_user #{username} #{password}").error?
+      converge_by "Updating password for user #{username} ..." do
+        shell_out!("rabbitmqctl update_user #{username} --password #{password}")
+      end
+    end
+  end
+end
+```
+
+## provides
+
+### Introduced
+
+Chef Infra Client <!-- TODO:  -->
+
+Use the `provides` method to associate multiple custom resource files with the same resources name
+For example:
+
+```ruby
+# Provide my_custom_resource to Redhat 7 and above
+provides :my_custom_resource, platform: 'redhat' do |node|
+  node['platform_version'].to_i >= 7
+end
+
+# Provide my_custom_resource to all Redhat platforms
+provides :my_custom_resource, platform: 'redhat'
+
+# Provide my_custom_resource to the RedHat platform family
+provides :my_custom_resource, platform_family: 'rhel'
+
+# Provide my_custom_resource to all linux machines
+provides :my_custom_resource, os: 'linux'
+
+# Provide my_custom_resource, useful if your resource file is not named the same as the resource you want to provide
+provides :my_custom_resource
+```
+
+This allows you to use multiple custom resources files that provide the
+same resource to the user, but for different operating systems or
+operation system versions. With this you can eliminate the need for
+platform or platform version logic within your resources.
+
+### Presidence
+
+Use the `provides` method to associate a custom resource with the Recipe
+DSL on different operating systems. When multiple custom resources use
+the same DSL, specificity rules are applied to determine the priority,
+from highest to lowest:
+
+1. provides :my_custom_resource, platform_version: '0.1.2'
+2. provides :my_custom_resource, platform: 'platform_name'
+3. provides :my_custom_resource, platform_family: 'platform_family'
+4. provides :my_custom_resource, os: 'operating_system'
+5. provides :my_custom_resource
+
+## reset_property
+
+Use the `reset_property` method to clear the value for a property as if
+it had never been set, and then use the default value. For example, to
+clear the value for a property named `password`:
+
+```ruby
+reset_property(:password)
+```
+
+## coerce
+
+`coerce` is used to transform user input into a canonical form. The
```

Comment:
"canonical form" is a really confusing way of describing what we're doing here

Reply:
Indeed. I'm not sure what that came from (definitely not a term I use regularly 😆 )